### PR TITLE
Add Kubernetes Access support to Machine ID

### PR DESCRIPTION
**Pull request by timothyb89**
This adds support for Kubernetes Access to Machine ID. Users can now request access to a Kubernetes cluster with the new `kubernetes_cluster` config field. When a request is configured, a `kubeconfig.yaml` is generated which can be used to access the given cluster.

As part of this, we also refactor a few parts of the bot code to help cache various requests that config templates make frequently by passing a new trivial `bot.B` interface to the `Render()` function. For good measure, we also fix the `destination.Destination` stutter by moving the interface to `bot.Destination`.

---

Users can request access to a Kubernetes cluster by adding a quick config field to their `tbot.yaml`:

```yaml
auth_server: proxy.xample.com:3080

onboarding:
  token: asdf1234
  ca_pins:
    - sha256:asdf1234

destinations:
  - directory: /opt/machine-id
    kubernetes_cluster: kubedemo
```

This will automatically generate a `kubeconfig.yaml` at `/opt/machine-id/kubeconfig.yaml`. You can then access the cluster:

```bash
KUBECONFIG=/opt/machine-id/kubeconfig.yaml kubectl get pods --all-namespaces
```

(the bot will need a role granting k8s access)

### CODE REVIEW

**Comment by timothyb89** [comment]
As an aside, I'd like to see https://github.com/gravitational/teleport/pull/13231 merged first to hopefully cut down on merge conflict shenanigans.

**Comment by strideynet** [comment]
Agreed, now I'm back from PTO, I'm going to chase review on #13231 to try and get that down.

File: lib/tbot/bot/bot.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/client/webclient"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+)
+
+// B is an interface covering various public Bot methods to circumvent
+// import cycle issues.
+type B interface {
```
Comment (strideynet) [inline lib/tbot/bot/bot.go:30]: Hmm, I'm always a little uncomfortable with shared interfaces like this. It's a pattern that's currently extensively used in the Teleport codebase, but goes against most of the Go fundementals I'm familiar with. It's preferable imo for each consumer to define its own interface, even if this results in some duplication.

**Review by strideynet — COMMENTED** [review]
One small comment on the shared interface, but the rest looks good. Nice work on including a unit test :)

**Review by espadolini — APPROVED** [review]
Is the CA caching required for this change, or was it just an unrelated improvement that was small enough to be merged as part of this?

**Review by strideynet — APPROVED** [review]

**Pull request merged**